src/Sylius/Bundle/ProductBundle/Doctrine/ORM/ProductOptionRepository.php

```diff
@@ -40,4 +41,40 @@ public function findByName(string $name, string $locale): array
             ->getResult()
         ;
     }
+
+    /**
+     * @return ProductOptionInterface[]
+     */
+    public function findByNamePart(string $phrase, string $locale, int $limit = 15): array
+    {
+        $expr = $this->getEntityManager()->getExpressionBuilder();
+
+        return $this->createQueryBuilder('o')
+            ->innerJoin('o.translations', 'translation', 'WITH', 'translation.locale = :locale')
+            ->andWhere($expr->orX(
+                'translation.name LIKE :phrase',
+                'o.code LIKE :phrase',
+            ))
+            ->setParameter('phrase', '%' . $phrase . '%')
+            ->setParameter('locale', $locale)
+            ->getQuery()
+            ->getResult()
+        ;
+    }
+
+    /**
+     * @param string[]|null $code
+     * @return ProductOptionInterface[]
+     */
+    public function findByOptions(?array $code = []): array
```

Comment:
```suggestion
    public function findByCodes(array $codes = []): array
```
Don't see the reason for allowing null